### PR TITLE
Backfill Tentacle-Deployment tests

### PR DESCRIPTION
**Pull request by flin-8**
[sc-70686]
This PR backfills some missing tests for the Tentacle deployment & the server-auth secret.

As part of writing these tests we discovered that really [long agent names don't get truncated correctly](https://app.shortcut.com/octopusdeploy/story/78448/long-target-name-doesn-t-install) - that will fixed later.

### CODE REVIEW

File: charts/kubernetes-agent/tests/tentacle-deployment_test.yaml [hunk 1]
```diff
@@ -129,3 +129,11 @@ tests:
...
             limits:
               cpu: "200m"
+              
+  - it: "truncates long names"
+    set:
+      nameOverride: thisisaverylongstring34087236548972364802367y408923604872341234141
+    asserts:
+      - equal:
+          path: metadata.name
+          value: thisisaverylongstring34087236548972364802367y408923604872341234-tentacle
```
Comment (flin-8) [inline charts/kubernetes-agent/tests/tentacle-deployment_test.yaml:139]: @APErebus is the name meant to be shorter than 63 chars?

Comment (APErebus) [reply charts/kubernetes-agent/tests/tentacle-deployment_test.yaml:139]: it should automatically truncate it yes

Comment (flin-8) [reply charts/kubernetes-agent/tests/tentacle-deployment_test.yaml:139]: I should've been more specific - do we expect the whole name to be truncated to 63 chars? right now we truncate then add `-tentacle` to the name (which brings it over 63 chars)

File: charts/kubernetes-agent/tests/tentacle-server-auth-secret_test.yaml [hunk 2]
```diff
@@ -2,6 +2,24 @@ suite: "tentacle configuration"
...
+    - matchSnapshot: {}
+- it: "Sets bearer token if specified"
+  set:
+    nameOverride: the-agent-name-lobster
+    agent:
+      bearerToken: fkadjsgfjkaeuwygfkuwyafgu3gwfkua36wgfu3wgkujafghj
+  asserts:
+  - equal:
+      path: data.bearer-token
+      value: ZmthZGpzZ2Zqa2FldXd5Z2ZrdXd5YWZndTNnd2ZrdWEzNndnZnUzd2drdWphZmdoag==
```
Comment (flin-8) [inline charts/kubernetes-agent/tests/tentacle-server-auth-secret_test.yaml:16]: Can we call base64enc from in here?

File: charts/kubernetes-agent/tests/tentacle-deployment_test.yaml (hunk 1 above)
Comment (APErebus) [reply charts/kubernetes-agent/tests/tentacle-deployment_test.yaml]: ooo, interesting

File: charts/kubernetes-agent/tests/tentacle-server-auth-secret_test.yaml (hunk 2 above)
Comment (APErebus) [reply charts/kubernetes-agent/tests/tentacle-server-auth-secret_test.yaml:16]: I don't know :|

**Review by APErebus — APPROVED** [review]
excellent! Thanks @flin-8

File: charts/kubernetes-agent/tests/tentacle-deployment_test.yaml (hunk 1 above)
Comment (flin-8) [reply charts/kubernetes-agent/tests/tentacle-deployment_test.yaml]: This potential bug is covered by https://app.shortcut.com/octopusdeploy/story/78448/long-target-name-doesn-t-install

**Pull request merged**